src/schedule/train_schedule.py

```diff
@@ -1,25 +1,55 @@
+import marshmallow as marsh
+from peewee import ForeignKeyField, IntegerField, TextField
+
+from src.base_model import BaseModel
 from src.schedule.schedule import Schedule
 
 
 class TrainSchedule(Schedule):
     """A schedule for spawning SUMO trains."""
 
-    @classmethod
-    def from_json(cls, json_object: str) -> "TrainSchedule":
-        """Constructs a TrainSchedule from a JSON object.
+    class Schema(Schedule.Schema):
+        """Schema for TrainSchedule."""
 
-        :param json_object: The JSON object.
-        :type json_object: str
-        :return: A TrainSchedule.
-        :rtype: TrainSchedule
-        """
-        raise NotImplementedError()
+        train_type = marsh.fields.Str(required=True)
+
+        def _make(self, data: dict) -> "TrainSchedule":
+            """Constructs a TrainSchedule object from a dictionary.
 
-    def maybe_spawn(self, tick: int):
-        """Determines whether a train should be spawned at the
-        current tick and spawns it if so.
+            :param data: the dictionary
+            :return: an instance of the model
+            """
+            return TrainSchedule(**data)
 
-        :param tick: The current tick.
-        :type tick: int
+    train_type = TextField(null=False)
+
+    def _spawn(self, traci_wrapper: "ITraCiWrapper"):
```

Comment:
I'm unsure if the name of ITraCIWrapper is "ITraCiWrapper" 🤔

Reply:
Resolved. See above.